src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -298,6 +297,9 @@ void prepare_primitive_fusing::fuse_bias(program &p) {
                 const_dep.get_output_layout().count() != static_cast<size_t>(out_features)) {
                 continue;
             }
+            // Handle eltw as a bias only when eltw's shape is to be broadcasted to parent node
+            if (const_dep.get_output_layout().count() > node->get_dependency(non_const_dep_idx).get_output_layout().count())
+                continue;
```

Comment:
I think it is necessary to compare each axis separately. For example, `conv_output_shape=[1,64,8,8]` and `eltw_shape=[64,1,1,1]`, I guess it is not broadcastable.

Reply:
At first I tried to add a check of  broadcastable or not. 
However the legacy shape (allow_no_shape_infer) is not good to work with bradcastability. 
For example your example also might be from [1, 64, 8, 8] and [64] where the 64 is expadnded to the legacy shape. 

2nd, checking size is enough becuase our jitter is supporting braodcast of fused input (const) to output of prev node, while not supporting broadcasting prev node output to fused node.

Reply:
As another example, current eltw is sometimes doing bad shape inferece for such a legacy case, such as 
fc output is 3d  [1, 512, 1024] and expanded to [1, 512, 1024, 1] for onednn case, 
and the const input is [1024] 
In this case, eltwise is setting the resulting shape as [1024, 512, 1024, 1]. 
However if the eltw is fused, fc output is again setting the output shape as original output shape, so the result is okay.  
This cannot be detected by broadcasting rule. I think this kind of issue should be modified by removing the legacy shape inference, which cannot be covered at this time.